### DISCUSSION TITLE
How to fix the Error "Cannot read propertoies of Null (Cannot read 'Iterations')"?

### DISCUSSION THREAD

**Discussion by CrunchyToasty** [dani-garcia/vaultwarden#3325] [Q&A]
Hello everyone,

Im running Vaultwarden at the moment on the Version 1.27.0 as an Docker Container.

My Problem is, that i only can log into Vaultwarden over the Website but when i try to login over the Browser extension or the Windows App i get the following error code:
"Cannot read propertoies of Null (Cannot read 'Iterations')"

I have Vaultwarden in use for ~1 Year, with an old Version. I didnt made changes to the configuration or the Enviroment and one day i got this error in the Bitwarden Apps. 

The first thing ive Done was creating a new Docker Container and importing my database and the error isnt fixed with that.

If you need more informations, please let me know.

Greets 
Marvin

**Comment by BlackDex** [comment]
Looks like you are not using HTTPS.
Or, not using the latest extension version maybe?

**Reply by CrunchyToasty** [reply]
Ive setup an reverse Proxy after this Tutorial.
https://www.wundertech.net/how-to-self-host-the-password-manager-bitwarden-on-a-synology-nas/

**Reply by CrunchyToasty** [reply]
The Destination Port is using HTTP, do i need to make changes to the Vaultwarden Config so its https?

**Reply by BlackDex** [reply]
Between the reverse-proxy and Vaultwarden no. But between the client and the reverse-proxy yes.

**Reply by CrunchyToasty** [reply]
Dani Garcia written in this wiki (Link below) that using an reverse Proxy is recommendet. So that shouldnt be the Problem.

I've tested the Browsers Firefox, Chrome and Edge with old and new Extentions and they all have the Problem.

 https://github-wiki-see.page/m/dani-garcia/vaultwarden/wiki/Enabling-HTTPS

**Reply by stefan0xC** [reply]
Are you sure you are using the latest release? Can you please provide the support string via the diagnostics page in the [`/admin` panel](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page).

**Reply by CrunchyToasty** [reply]
Sure, but i need a moment to do that.

**Reply by BlackDex** [reply]
I juist quickly tested locally with the v2023.2.0 extension on both FireFox and Chromium.
I can connect just fine using both HTTP and HTTPS.

**Reply by CrunchyToasty** [reply]
I have an Problem with enabling the admin token. Im not sure how to set the variabel for the mounting path right. 
Im running docker on my Synology NAS on my Volume 2 and there is an Folder for Docker and inside that is my Btiwarden Folder.

**Reply by BlackDex** [reply]
mounting path for the admin token??
I'm not sure what you mean with that.
I suggest reading: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page

**Reply by CrunchyToasty** [reply]
I got the Problem with this Variable:  -v /vw-data/:/data/ \
Ive read this before: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page

**Reply by BlackDex** [reply]
Please read:
- https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container
- https://github.com/dani-garcia/vaultwarden/wiki/Using-Docker-Compose

In regards to using volumes. Make sure that the `/vw-data/` directory exists on the host and is accessible by docker.

**Reply by CrunchyToasty** [reply]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.27.0
* Web-vault version: v2022.12.0
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: false
* HTTPS Check: false
* Database type: SQLite
* Database version: 3.39.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "****://*********",
  "domain_origin": "****://*********",
  "domain_path": "",
  "domain_set": false,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Reply by CrunchyToasty** [reply]
Ive restarted the new Vaultwarden Container without the admin page and now its working again...

**Comment by keneichi** [comment]
Hi,

I use bitwarden_RS (old name of vaultwarden) for 3 or 4 years I guess.
I use Opera, Firefox and chrome plugin without any problem. My server is behind a Nginx reverse proxy and i use let's encrypt SSL Certificate and MFA.
Yesterday i tried an another web brower called SideKick (based on Chromium). And for the first time I met the same problem "Cannot read propertoies of Null (Cannot read 'Iterations')".

Plugin on other browsers still working fine.

Vincent

**Reply by BlackDex** [reply]
If using Bitwarden_RS then you **MUST** update a.s.a.p. All the latest clients will **not** work anymore with that version.
Also, many performance improvements and features are with the newer versions. Also, we do **not** support Bitwarden_RS anymore.

**Reply by keneichi** [reply]
Thanks for the advice. It was a project i had... i know now i have to do this quickly :)

**Reply by keneichi** [reply]
I just finish the migration into vaultwarden... 
no more problem on any brower...
